Source/Scene/GroundPolylinePrimitive.js

```diff
@@ -0,0 +1,616 @@
+            }
+
+            primitiveOptions.geometryInstances = polylineSegmentInstances;
+            primitiveOptions.appearance = this.appearance;
+
+            primitiveOptions._createShaderProgramFunction = function(primitive, frameState, appearance) {
+                createShaderProgram(that, frameState, appearance);
+            };
+            primitiveOptions._createCommandsFunction = function(primitive, appearance, material, translucent, twoPasses, colorCommands, pickCommands) {
+                createCommands(that, appearance, material, translucent, colorCommands, pickCommands);
+            };
+            primitiveOptions._updateAndQueueCommandsFunction = function(primitive, frameState, colorCommands, pickCommands, modelMatrix, cull, debugShowBoundingVolume, twoPasses) {
+                updateAndQueueCommands(that, frameState, colorCommands, pickCommands, modelMatrix, cull, debugShowBoundingVolume);
+            };
+
+            this._primitive = new Primitive(primitiveOptions);
+            this._primitive.readyPromise.then(function(primitive) {
+                that._ready = true;
+
+                if (that.releaseGeometryInstances) {
+                    that.polylineGeometryInstances = undefined;
+                }
+
+                var error = primitive._error;
+                if (!defined(error)) {
+                    that._readyPromise.resolve(that);
+                } else {
+                    that._readyPromise.reject(error);
+                }
+            });
+        }
+        this._primitive.appearance = this.appearance;
+        this._primitive.show = this.show;
+        this._primitive.debugShowBoundingVolume = this.debugShowBoundingVolume;
+        this._primitive.update(frameState);
+    };
+
+    GroundPolylinePrimitive.prototype.isDestroyed = function() {
+        return false;
+    };
+
+    GroundPolylinePrimitive.prototype.destroy = function() {
+        this._primitive = this._primitive && this._primitive.destroy();
+        this._sp = this._sp && this._sp.destroy();
+        this._spPick = this._spPick && this._spPick.destroy();
+
+        // Derived programs, destroyed above if they existed.
+        this._sp2D = undefined;
+        this._spPick2D = undefined;
+
+        //These objects may be fairly large and reference other large objects (like Entities)
+        //We explicitly set them to undefined here so that the memory can be freed
+        //even if a reference to the destroyed GroundPolylinePrimitive has been kept around.
+        this._idsToInstanceIndices = undefined;
+        this._attributeSynchronizerCache = undefined;
+
+        return destroyObject(this);
+    };
+
+    // An object that, on setting an attribute, will set all the instances' attributes.
+    var userAttributeNames = ['width', 'show'];
+    var userAttributeCount = userAttributeNames.length;
+    function InstanceAttributeSynchronizer(batchTable, firstInstanceIndex, lastInstanceIndex, batchTableAttributeIndices) {
```

Comment:
Once you remove the per segment data from the batch table, you can remove this too.

Reply:
You could also move most of the code from `GroundLineSegmentGeometry` to `GroundPolylineGeometry` and remove `decompose`.